### ISSUE TITLE
Spies do not register as allowed method calls when stubbing is specified

### ISSUE THREAD

**Issue by ciaranmcnulty** [phpspec/prophecy#120]
Normally Spied-on methods are included in the list of expected calls. However, this breaks if stubbing takes place beforehand

Spec:

``` php
class SpiesExampleSpec extends ObjectBehavior
{
    function it_does_not_expect_spied_on_methods(\DateTime $time)
    {
        $time->getOffset()->willReturn(-1800);

        $this->modifyTimestamp($time);

        $time->setTimestamp(1234)->shouldHaveBeenCalled();
    }
}
```

Object:

``` php
class SpiesExample
{
    public function modifyTimestamp(\DateTime $time)
    {
        $time->getOffset();
        $time->setTimestamp(1234);
    }
}
```

Output:

```
  10  ! it does not expect spies
      method call:
        Double\DateTime\P1->setTimestamp(1234)
      was not expected.
      Expected calls are:
        - getOffset()
```

**Comment by MPV** [comment]
I just ran in to this issue as well (I have a stub that I expect a method to be called on, but `shouldHaveBeenCalled()` gives the error that @ciaranmcnulty mentioned).

**Comment by scaytrase** [comment]
Any news on this bug here in 2019?

**Comment by jdreesen** [comment]
Can this be closed, now that #441 is merged?
